src/streaming.ts

```diff
@@ -46,46 +52,96 @@ export class StreamProcessor {
     this.eventBus = eventBus;
     this.repository = repository;
     this.log = this.options.logger;
+
+    const minDelay = 5000;
+    const maxDelay = 10000;
+    this.retryAttempt = 1;
+    this.retryDelayMs = Math.floor(Math.random() * (maxDelay - minDelay + 1) + minDelay)
   }
 
   start(): void {
-    this.log.info('Starting StreamProcessor');
-    const eventSource = new EventSource(
-      `${this.options.baseUrl}/stream?cluster=${this.cluster}`,
-      {
-        headers: {
-          Authorization: `Bearer ${this.jwtToken}`,
-          'API-Key': this.apiKey,
-        },
-      },
-    );
-
-    eventSource.onopen = (event: MessageEvent) => {
-      this.log.debug('Stream connected', event);
+    this.log.info('Starting new StreamProcessor');
+
+    const url = `${this.options.baseUrl}/stream?cluster=${this.cluster}`
```

Comment:
Does this section need to exist? it looks like the openapi generated client already has a helper to call the `stream` endpoint. But that also begs the question of why this wasn't used instead of `eventsource`
https://github.com/harness/ff-nodejs-server-sdk/blob/3388147168527e77abcc84bb2f29d4e62e0f7814/src/openapi/api.ts#L1078

Reply:
Without knowing the historical context of the code, my guess is the openapi spec includes this and generates the relevant code but we weren't using it anyhow (instead we had eventsource), possibility due to headers needing added or due to the SSE socket needing to live for a long time, unlike regular rest requests. Some of the openapi generators can be quiet buggy too (for other language).